stable/jenkins/templates/_helpers.tpl

```diff
@@ -53,3 +53,14 @@ Create the name of the service account to use
     {{ default "default" .Values.serviceAccount.name }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Create the name of the service account for Jenkins agents to use
+*/}}
+{{- define "jenkins.serviceAccountAgentName" -}}
+{{- if .Values.serviceAccountAgent.create -}}
+    {{ default (include "jenkins.fullname" .) .Values.serviceAccountAgent.name }}
```

Comment:
I think we have a problem here. If neither` Values.serviceAccount.name` nor `.Values.serviceAccountAgent.name` is set then we are trying two service accounts with the same name.

```
jenkins.serviceAccountName: 
{{ default (include "jenkins.fullname" .) .Values.serviceAccount.name }}

jenkins.serviceAccountAgentName
{{ default (include "jenkins.fullname" .) .Values.serviceAccountAgent.name }}
```

We could add the "-agent" suffix:
```
{{ default (printf "%s-%s" (include "jenkins.fullname" .) "agent") .Values.serviceAccountAgent.name }}
```

Reply:
Good point. I thought there might have been a collision somewhere but I was hung up on there being two different bindings to default but that's not an issue.